diff --git a/binding.gyp b/binding.gyp
index bb40aa7e8d41644ba690761e9064fc1d0f2dcd6b..73bf0ae198265baea3bca10d36b9e211f140be58 100644
--- a/binding.gyp
+++ b/binding.gyp
@@ -25,13 +25,13 @@
         "NAPI_VERSION=<(napi_build_version)",
       ],
       "cflags_cc": [
-        "-std=c++17"
+        "-std=gnu++20"
       ],
       "conditions": [
         ["OS=='mac'", {
           "xcode_settings": {
             "GCC_SYMBOLS_PRIVATE_EXTERN": "YES", # -fvisibility=hidden
-            "CLANG_CXX_LANGUAGE_STANDARD": "c++17",
+            "CLANG_CXX_LANGUAGE_STANDARD": "gnu++20",
             "MACOSX_DEPLOYMENT_TARGET": "10.9",
           },
         }],
